### PR TITLE
feat: add methods to get page numbers in PagerRenderer

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Supersedes #4141

Add methods to get page numbers in PagerRenderer.
To generate Pager HTML that is the same as CI3's.

- getFirstPageNumber()
- getCurrentPageNumber()
- getLastPageNumber()
- getPreviousPageNumber()
- getNextPageNumber()

The template would be like this:
```php
<?php if ($pager->hasPreviousPage()) : ?>
<a href="<?= $pager->getFirst() ?>" data-ci-pagination-page="<?= $pager->getPageNumberFirst() ?>" rel="start">&laquo;First</a>
<a href="<?= $pager->getPreviousPage() ?>" data-ci-pagination-page="<?= $pager->getPreviousPageNumber() ?>" rel="prev">&lt;</a>
<?php endif ?>
```

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by paulbalandan — COMMENTED** [review]
How about making it more sound natural? Like getFirstPageNumber, getCurrentPageNumber, getLastPageNumber. Then add another for getPreviousPageNumber.

Or better yet, remove the Number? getFirstPage, getLastPage etc.

**Comment by michalsn** [comment]
> Or better yet, remove the Number? getFirstPage, getLastPage etc.

We already have methods in this convention and they all return an URL, so it might be confusing for users. But your idea of having getFirstPageNumber, getCurrentPageNumber, getLastPageNumber sounds better to me - more natural.

@kenjis  One thing I don't like is this:
```php
$pager->getPageNumberCurrent() - 1
```
We should add a method `getPreviousPageNumber()` (and `getNextPageNumber()`) that will make calculations for us. Of course, we still can have `getCurrentPageNumber()` method.

**Comment by kenjis** [comment]
I changed the method names, and added `getPreviousPageNumber()` and `getNextPageNumber()`.

File: tests/system/Pager/PagerRendererTest.php [hunk 1]
```diff
@@ -560,4 +560,103 @@ public function testGetNextPageWithSegmentHigherThanZero()
...
+			'uri'         => $uri,
+			'pageCount'   => 10,
+			'currentPage' => 1,
+			'total'       => 100,
+			'segment'     => 2,
+		];
+		$pager   = new PagerRenderer($details);
+		$pager->setSurroundCount(2);
+
+		$this->assertEquals(null, $pager->getPreviousPageNumber());
```
Comment (paulbalandan) [inline tests/system/Pager/PagerRendererTest.php:628]:
```suggestion
		$this->assertNull($pager->getPreviousPageNumber());
```

File: tests/system/Pager/PagerRendererTest.php [hunk 2]
```diff
@@ -560,4 +560,103 @@ public function testGetNextPageWithSegmentHigherThanZero()
...
+			'uri'         => $uri,
+			'pageCount'   => 10,
+			'currentPage' => 10,
+			'total'       => 100,
+			'segment'     => 2,
+		];
+		$pager   = new PagerRenderer($details);
+		$pager->setSurroundCount(2);
+
+		$this->assertEquals(null, $pager->getNextPageNumber());
```
Comment (paulbalandan) [inline tests/system/Pager/PagerRendererTest.php:660]:
```suggestion
		$this->assertNull($pager->getNextPageNumber());
```

**Review by paulbalandan — APPROVED** [review]

**Review by MGatner — APPROVED** [review]
Looks good! I think this is a better solution than the magic getter. Please implement @paulbalandan 's assertion changes, then I think we're set.

**Comment by kenjis** [comment]
@MGatner Done! Thanks for all.

**Review by MGatner — APPROVED** [review]

**Pull request merged**